src/audio/copier/copier.c

```diff
@@ -1716,18 +1716,25 @@ static uint64_t copier_get_processed_data(struct comp_dev *dev, uint32_t stream_
 {
 	struct copier_data *cd = comp_get_drvdata(dev);
 	uint64_t ret = 0;
-	bool source = dev->direction == SOF_IPC_STREAM_CAPTURE;
 
-	if (cd->endpoint_num && cd->bsource_buffer != input) {
-		if (stream_no < cd->endpoint_num) {
-			if (dev->ipc_config.type == SOF_COMP_HOST && !cd->ipc_gtw) {
-				if (source == input)
-					ret = cd->hd->total_data_processed;
-			} else {
-				ret = comp_get_total_data_processed(cd->endpoint[stream_no],
-								    0, input);
-			}
+	/*
+	 * bsource_buffer is assigned in create host/dai/gtw, however, host/dai
+	 * part are removing this flag, so this condition is not valid, remove
+	 * cd->bsource_buffer != input.
+	 * Also, cd->endpoint_num currently are always 1, and stream_no always 0
+	 * no need to check this part.
+	 */
+	if (cd->endpoint_num) {
```

Comment:
Ok, so this is a good commit message and describes the patch code change.
But, the inline comment needs to describe the code and not the change. e.g.
```c
/* total data processed is calculated as follows ... describe calculation logic */
```